util/daemonize.c

```diff
@@ -7,6 +7,89 @@
 
 #include "daemonize.h"
 
+/* Read one byte from ${fd}, looping upon EINTR.  Do not store that byte. */
+static int
+readbyte(int fd)
+{
+	char dummy;
+	char done = 0;
+
+	do {
+		switch (read(fd, &dummy, 1)) {
+		case -1:
+			/* Anything other than EINTR is bad. */
+			if (errno != EINTR) {
+				warnp("read");
+				goto err0;
+			}
+
+			/* Otherwise, loop and read again. */
+			break;
+		case 0:
+			warn0("Unexpected EOF in pipe");
+			goto err0;
+		case 1:
+			/* Expected value; good, quit the loop. */
+			done = 1;
+		}
+	} while (!done);
+
+	/* Success! */
+	return (0);
+
+err0:
+	/* Failure! */
+	return (-1);
+}
+
+/* Send a byte to ${fd}, ignoring EINTR. */
+static int
+sendbyte(int fd)
```

Comment:
I added this as a `MAYBE???` commit.

I'd actually prefer to include this change, but I'm betting that you'll say that the (source-code) overhead of `sendbyte()` isn't worth it.

Reply:
Yeah, not worth it.